mlrun/artifacts/base.py

```diff
@@ -967,6 +968,11 @@ def spec(self, spec):
         self._spec = self._verify_dict(spec, "spec", LinkArtifactSpec)
 
 
+# TODO: remove in 1.5.0
+@deprecated(
```

Comment:
I believe the `@deprecated` decorator automatically sets a `DeprecationWarning` warning, and we would like `FutureWarning` since we want the end-user to see it. There should be a way to pass the type of warning category to the decorator (I think it's a `category=` parameter??)

Reply:
reverted back to `DeprecationWarning` because we filter `FutureWarning` and it needs to be investigated

Reply:
back to `FutureWarning` after complications with `DeprecationWarning` only showing some of the times